src/devices/rpLidar2/sdk_original/app/frame_grabber/drvlogic/lidarmgr.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ *  RPLIDAR
+ *  Win32 Demo Application
+ *
+ *  Copyright (c) 2009 - 2014 RoboPeak Team
+ *  http://www.robopeak.com
+ *  Copyright (c) 2014 - 2016 Shanghai Slamtec Co., Ltd.
+ *  http://www.slamtec.com
+ *
+ */
+/*
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "stdafx.h"
+#include "lidarmgr.h"
+
+LidarMgr * LidarMgr::g_instance = NULL;
+rp::hal::Locker LidarMgr::g_oplocker;
+RPlidarDriver * LidarMgr::lidar_drv = NULL;
+
+LidarMgr & LidarMgr::GetInstance()
+{
+    if (g_instance) return *g_instance;
+    rp::hal::AutoLocker l(g_oplocker);
+
+    if (g_instance) return *g_instance;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identical condition 'g_instance', second condition is always false](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5162834)